samcli/commands/package/package_context.py

```diff
@@ -79,9 +82,17 @@ def __exit__(self, *args):
 
     def run(self):
 
+        gc = GlobalConfig()
         session = boto3.Session(profile_name=self.profile if self.profile else None)
         s3_client = session.client(
-            "s3", config=Config(signature_version="s3v4", region_name=self.region if self.region else None)
+            "s3",
+            config=Config(
+                signature_version="s3v4",
+                region_name=self.region if self.region else None,
+                user_agent_extra=f"aws-sam-cli/{__version__}/{gc.installation_id}"
```

Comment:
can we move this to utils? don't want to duplicate

Reply:
Good call out.